course3/homework-unique_ptr/unique_ptr.h

```diff
@@ -0,0 +1,39 @@
+#pragma once
+
+#include <cstddef>
+
+
+namespace karun
+{
+	template <class T>
+	class unique_ptr
+	{	
+		
+	public:
+		unique_ptr();
+		unique_ptr(std::nullptr_t);
```

Comment:
http://en.cppreference.com/w/cpp/language/nullptr

> There exist implicit conversions from nullptr to null pointer value of any pointer type

I don't think the overload for nullptr_t is required at all.